src/main/java/domain/Ladder.java

```diff
@@ -1,30 +1,38 @@
 package domain;
 
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import util.Errors;
 
 public class Ladder {
 
-    private final List<Boolean> rungs;
+    private final List<Line> lines;
 
-    private Ladder(List<Boolean> rungs) {
-        this.rungs = new ArrayList<>(rungs);
+    public Ladder(List<Line> lines) {
+        validate(lines);
+        this.lines = new ArrayList<>(lines);
     }
 
-    public static Ladder from(List<Boolean> rungsStatus) {
-        return new Ladder(rungsStatus);
+    private static void validate(List<Line> lines) {
+        validateLaddersHeight(lines);
     }
 
-    public int getHeight() {
-        return rungs.size();
+    private static void validateLaddersHeight(List<Line> lines) {
+        boolean allSameHeight = lines.stream()
+            .map(Line::getHeight)
+            .distinct()
+            .count() == 1;
+        if (!allSameHeight) {
+            throw new IllegalArgumentException(Errors.ALL_LINE_MUST_HAVE_SAME_HEIGHT);
+        }
     }
 
-    public List<Boolean> getRungsStatus() {
-        return Collections.unmodifiableList(rungs);
+    public List<List<Boolean>> getRightRungStatus() {
+        List<List<Boolean>> rightRungStatus = new ArrayList<>();
+        for (Line line : lines) {
+            rightRungStatus.add(line.getRightStatus());
+        }
+        return rightRungStatus;
```

Comment:
```suggestion
    public List<List<Boolean>> getRightRungStatus() {
        return lines.stream()
            .map(Line::getRightStatus)
            .toList();
    }
``` 
참고사항인데 stream을 통해서 더 간편하게 나타낼 수 있을 것 같습니다.